spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,111 @@
+RSpec.describe PasswordResetsController do
+  describe "POST #create" do
+    let(:user)      { create(:confirmed_user) }
+    let(:params)    { { password_reset: { email: user.email } } }
+    let(:arguments) { { email: user.email } }
+    let(:context)   { double(:context, success?: true) }
+
+    before(:example) do
+      allow(PasswordResetOrganizer).to receive(:call).with(arguments).
+        and_return(context)
+    end
+
+    it "calls PasswordResetOrganizer" do
+      expect(PasswordResetOrganizer).to receive(:call)
+      post :create, params
+    end
+
+    context "when successful" do
+      it "returns HTTP status 201" do
+        post :create, params
+        expect(response).to have_http_status(201)
+      end
+
+      it "redirects to index" do
+        post :create, params
+        expect(response).to redirect_to("/")
+      end
+
+      it "renders a success notice" do
+        post :create, params
+        expect(JSON.parse(response.body)).to eq(password: ["reset sent"])
+      end
+    end
+
+    context "when unsuccessful" do
+      let(:context) { double(:context, success?: false) }
+
+      it "returns HTTP status 404" do
+        post :create, params
+        expect(response).to have_http_status(404)
+      end
+
+      it "redirects to index" do
+        post :create, params
+        expect(response).to redirect_to("/")
+      end
+
+      it "renders an error" do
+        post :create, params
+        expect(JSON.parse(response.body)).to eq(password: ["reset failed"])
+      end
+    end
+  end
+
+  describe "PATCH #udpate" do
+    let(:user)      { create(:confirmed_user) }
+    let(:params)    { { password_reset: { email: user.email } } }
+    let(:arguments) { { user_params: user_params, user: user } }
+    let(:context)   { double(:context, success?: true) }
+
+    let(:user_params) do
+      { password: user.password }
+    end
+
+    before(:example) do
+      allow(UpdatePassword).to receive(:call).with(arguments).
+        and_return(context)
+    end
+
+    it "calls UpdatePassword interactor" do
+      expect(UpdatePassword).to receive(:call)
+      patch :update, params
+    end
+
+    context "when successful" do
+      it "returns HTTP status 200" do
+        patch :update, params
+        expect(response).to have_http_status(200)
+      end
+
+      it "redirects to /signin" do
+        patch :update, params
+        expect(response).to redirect_to("/signin")
```

Comment:
Knock is built as a `Rails Engine`. It's basically a mini Rails app that cannot run on its own. It has to be "mounted" onto a full Rails app. The way it works lets you choose what the endpoint in which the engine gets mounted is. You could make that whatever you like, `/login`, `/knockin`, `/enrique/is/super/awesome`, etc. If you look at `config/routes.rb` you will see where the engine was mounted.

So... no, it's not a reserved route :smile:. 

Also, you don't have a to create a corresponding `/signin` view in the front-end. Or better said, you don't have to match the endpoints. Your users will need a way to sign in for sure but that endpoint is independent from your API's endpoint. On the front-end with some JavaScript magic, you are simply hitting your API and rendering the results to your users.